src/main/java/frc/robot/maps/FrancoisMap.java

```diff
@@ -88,6 +97,38 @@ public PIDSparkMax elevator() {
 
                 return new PIDSparkMax(leader);
             }
+
+            @Override
+            public ISolenoid liftBrake() {
+                // TODO Change this to a real Channel
+                WSolenoid brake = new WSolenoid(3);
+
+                return brake;
+            }
+
+            @Override
+            public BooleanSupplier upperLiftLimit() {
+                // TODO Change this to a real Channel
+                InvertDigitalInput upperLimit = new InvertDigitalInput(0);
+
+                return upperLimit::get;
+            }
+
+            @Override
+            public BooleanSupplier lowerLiftLimit() {
+                // TODO Change this to a real Channel
+                InvertDigitalInput lowerLimit = new InvertDigitalInput(1);
+
+                return lowerLimit::get;
+            }
+
+            @Override
+            public IEncoder getLiftEncoder() {
```

Comment:
This encoder is really coming from the `CANSparkMax`. We probably need to define `leader` outside `elevator()` and then return the `SparkMaxEncoder` object here. Though we may not actually need the separate encoder at all if we get the PID working correctly.

Reply:
wait what?

Reply:
The `TODO` can be removed and you can skip the intermediate variable.